testsuite/integration/elytron/src/test/java/org/wildfly/test/integration/elytron/authmode/ConstraintDrivenAuthModeTestCase.java

```diff
@@ -0,0 +1,167 @@
+                .build()) {
+            try (CloseableHttpResponse response = httpClient.execute(request)) {
+                int statusCode = response.getStatusLine().getStatusCode();
+                assertEquals("Unexpected status code in HTTP response.", SC_UNAUTHORIZED, statusCode);
+            }
+        }
+    }
+
+    @Test
+    public void testUnsecureResourceWithInvalidCredential() throws Exception {
+        HttpGet request = new HttpGet(new URI(url.toExternalForm() + "unsecure.jsp"));
+        UsernamePasswordCredentials credentials = new UsernamePasswordCredentials("user1", "password2");
+
+        CredentialsProvider credsProvider = new BasicCredentialsProvider();
+        credsProvider.setCredentials(AuthScope.ANY, credentials);
+        request.addHeader(new BasicScheme().authenticate(credentials, request));
+
+        try (CloseableHttpClient httpClient = HttpClients.custom()
+                .build()) {
+            try (CloseableHttpResponse response = httpClient.execute(request)) {
+                int statusCode = response.getStatusLine().getStatusCode();
+                assertEquals("Unexpected status code in HTTP response.", SC_OK, statusCode);
+                assertEquals("Unexpected content of HTTP response.", "", EntityUtils.toString(response.getEntity()));
+            }
+        }
+    }
+
+    @Test
+    public void testUnsecureResourceWithoutCredential() throws Exception {
+        HttpGet request = new HttpGet(new URI(url.toExternalForm() + "unsecure.jsp"));
+
+        try (CloseableHttpClient httpClient = HttpClients.custom().build()) {
+            try (CloseableHttpResponse response = httpClient.execute(request)) {
+                int statusCode = response.getStatusLine().getStatusCode();
+                assertEquals("Unexpected status code in HTTP response.", SC_OK, statusCode);
+                assertEquals("Unexpected content of HTTP response.", "", EntityUtils.toString(response.getEntity()));
+            }
+        }
+    }
+
+    static class ServerSetup extends AbstractElytronSetupTask {
+
+
+        @Override
+        protected ConfigurableElement[] getConfigurableElements() {
+            ConfigurableElement[] configurableElements = new ConfigurableElement[1];
+            configurableElements[0] = new ConfigurableElement() {
+                @Override
+                public String getName() {
+                    return "Enable CONSTRAINT-DRIVEN auth";
+                }
+
+                @Override
+                public void create(CLIWrapper cli) throws Exception {
+                    cli.sendLine("/subsystem=undertow/servlet-container=default:write-attribute(name=proactive-authentication, value=false)");
+                }
+
+                @Override
+                public void remove(CLIWrapper cli) throws Exception {
+                    cli.sendLine("/subsystem=undertow/servlet-container=default:remove-attribute(name=proactive-authentication)");
```

Comment:
There's no 'remove-attribute' op. Did you mean 'undefine-attribute'?

Reply:
Yes sorry. Updated the PR now